specification/requirements.adoc

```diff
@@ -1,5 +1,167 @@
+CoVE implementation must meet software and hardware requirements, as described
+in the next sections.
+
+==== Ownership
+
+In a CoVE-IO context, the non-confidential host software stack is the sole
+platform resources owner. In particular, the physical devices to which TDIs
+belong are owned and managed by the host, typically through the device PF.
+Host software stack components, e.g. the VMM, assign, unassign and expose
+TDIs resp. to, from and to TVMs.
+
+==== Platform Hardware Components
+
+===== PCIe
+
+All CoVE-IO-compliant devices in a CoVE platform must be connected to the CoVE
+platform through a PCIe Root Port (RP) that is part of a PCIe Root Complex (RC).
+
+A CoVE-IO PCIe link can be direct or go via any topology or PCIe switches and
+bridges. Since those are excluded from a TVM's TCB, only selective IDE streams
+must be used between a CoVE-IO device and its corresponding PCIe Root Port.
+
+The TSM establishes one single selective IDE stream for each physical device
+from which TDIs may be attached to TVMs. All TDIs within a CoVE-IO device share
+the same IDE stream.
+
+For a given selective IDE stream, the TSM generates, owns and distributes the
+IDE keys to both the CoVE-IO device and its upstream PCIe Root Port.
+
+[[coveio_hierarchy]]
+.CoVE-IO PCIe Topology
+image::coveio_hierarchy.svg[align="center"]
+
+
+===== IO Translation Agent
+
+CoVE-IO-compliant platforms must have at least one IO translation agent, also
+known as an IOMMU. Platform IOMMUs must follow the RISC-V <<IOMMU>> architecture
+specification v1.0.
+
+In order to secure direct acces to confidential memory access, all inbound
+traffic from a TDI must be translated IOMMU translated.
+
+===== Hardware Root-of-Trust
+
+As described in <<PCIe>>, the TSM generates and sets the IDE keys into both the
+CoVE-IO PCIe endpoint and its upstream Root Port, for all maintained selective
+IDE streams.
+
+When setting IDE keys into a CoVE-IO device, the TSM relies on the DSM IDE Key
+Management (`IDE_KM`) support, and its ability to receive IDE_KM messages over a
+Secured SPDM session. However, there are no architecturally-defined PCIe
+protocol for managing Root Port IDE keys.
+
+Instead of adding multiple vendor-specific `IDE_KM` implementations to the TSM,
+the TSM relies on the platform hardware Root-of-Trust (HROT) to implement the
+`IDE_KM` protocol and abstract the platform specific PCIe RP implementation away
+from the TSM. The TSM establishes a Secured SPDM session with the HROT over a
+host accessible DOE mailbox, and then sets platform RP IDE keys over that
+session.
+
+As a consequence, a CoVE-IO-compliant platform must have at least one PCIe
+accessible HROT, with the following requirements:
+
+1. The HROT must support the DOE mechanism
+2. The HROT must support Secured SPDM sessions
+3. The HROT must support the IDE Key Management protocol
+
+===== CoVE-IO Manifest
+
+==== Software
+CoVE
+TSM (SPDM, TDISP)
+
+=== Guest
+
+CoVE-IO, TDM
```

Comment:
What is TDM ?

Reply:
Yes, I'll expand the glossary once I get to that part of the requirements.